application/src/main/java/org/thingsboard/server/service/subscription/TbEntityLocalSubsInfo.java

```diff
@@ -137,8 +135,30 @@ public TbEntitySubEvent remove(TbSubscription<?> sub) {
             return toEvent(ComponentLifecycleEvent.DELETED);
         }
         TbSubscriptionsInfo oldState = state.copy();
-        TbSubscriptionsInfo newState = new TbSubscriptionsInfo();
+
+        //copy unchanged state only
+        TbSubscriptionsInfo newState = state.copy();
```

Comment:
Remove all code that is not related to the scope of changes. Create new PR and cover this with TESTS!!!